scanner/indexer/indexer.go

```diff
@@ -571,6 +590,35 @@ func createManifestDigest(hashID string) (claircore.Digest, error) {
 	return d, nil
 }
 
+func (i *localIndexer) StoreIndexReport(ctx context.Context, hashID string, scannerVersion string, report *claircore.IndexReport) error {
+	ctx = zlog.ContextWithValues(ctx, "component", "scanner/backend/indexer.StoreIndexReport")
+
+	sv, err := pkgscanner.DecodeVersion(scannerVersion)
+	if err != nil {
+		return err
+	}
+	// Ensure the version strings contain valid semantic versions. This is
+	// important for comparability.
+	if !semver.IsValid(sv.Indexer) {
+		return fmt.Errorf("indexer version %q is not a valid semantic version", sv.Indexer)
+	}
+
+	err = i.externalIndexStore.StoreIndexReport(ctx, hashID, sv.Indexer, report, i.randomExpiry(time.Now()),
+		func(iv string) bool {
+			// If the stored indexer version is not valid, let the datastore
+			// overwrite the record.
+			v, err := pkgscanner.DecodeVersion(iv)
+			return err != nil ||
+				!semver.IsValid(v.Indexer) ||
+				semver.Compare(sv.Indexer, v.Indexer) >= 0
+		})
+	if err != nil {
+		return fmt.Errorf("storing external index report with (hashID %q): %w", hashID, err)
+	}
+
+	return nil
```

Comment:
Will need a way to reflect the 'status' of the store action (if still considering that for this phase), so that it can be returned in the response.